### ISSUE TITLE
[UI] Lynx-style jump to and highlight links

### ISSUE THREAD

**Issue by stroobandt** [qutebrowser/qutebrowser#2067]
The [`lynx`](https://en.wikipedia.org/wiki/Lynx_%28web_browser%29) text browser has a very efficient feature that I particularly like, 
but that appears to be completely absent in `qutebrowser`.

In Lynx, hitting <kbd>j</kbd> and <kbd>k</kbd>, or optionally, <kbd>↓</kbd> and <kbd>↑</kbd>, will jump from one link anchor to the next, highlighting it (in yellow). Hitting <kbd>Enter</kbd> suffices to open the link.

From a user experience perspective, such option may be more efficient and intuitive than <kbd>f</kbd> (which I do not condemn!) The user would save the mental burden of reading the <kbd>f</kbd> link letter(s) in small print and reproducing those on the keyboard.

> Browsing in Lynx consists of highlighting the chosen link using cursor keys, or having all links on a page numbered and entering the chosen link's number.

Related issue: https://github.com/The-Compiler/qutebrowser/issues/1453

Seen in version `0.8.2`.

**Comment by The-Compiler** [comment]
FWIW I fail to see how this would be more efficient than using `f` and typing something like two letters, for any page with more than a handful of links :wink:

**Comment by stroobandt** [comment]
What can I say?

- First of all, and by all means, give [`lynx`](https://en.wikipedia.org/wiki/Lynx_(web_browser)) a little try-out spin. That way, you will get a feel of what I mean.
- You can call me lazy if you want, but with `f` I need to read, two characters, interpret those in my brains and consequently reproduce them on the keyboard… That is relying on a lot of brain power when compared to **hitting <kbd>↓</kbd> until the link that one wants lights up.**
- I want to be able to use `qutebrowser` at 2AM in the morning, under dim lighting conditions and, being slightly tipsy…
- The disabled and the wankers among us will appreciate single handed browsing…
- You are still young. Revisit this post in twenty years. The whole `f` scheme might be confusing for the elderly or people with poor eye sight.

_Hope you could appreciate this Saturday morning humour, over there in Winterthur!_

**Comment by The-Compiler** [comment]
I'm well aware of how lynx (and the feature you mentioned) works :wink:

FWIW if someone comes up with a good PR I'd be fine with taking this in, but I'm unlikely to work on this myself, sorry.

Some other ideas:

- You can change the hint mode (`hints -> mode`) to `number` so you can type substrings of links to filter them, or to `word` if you find words easier to type.
- You can change `hints -> chars` for one-handed operation if you prefer
- You can change the hint font size in `fonts -> hints`

**Comment by The-Compiler** [comment]
So I just realized this is basically already possible, using `:set input spatial-navigation true`. The highlights aren't very visible, but that'll change with the upcoming QtWebEngine backend.

All that's left is making that a separate mode - see #609.

**Comment by stroobandt** [comment]
@The-Compiler Actually, I did try out `spatial-navigation` before filing this request. However, there are a small but significant number of differences with `lynx`.

With `lynx`, all navigation from link to link (or input field) is done with **only two keys:** <kbd>↑</kbd> and <kbd>↓</kbd> or `j` and `k` for that matter. Furthermore, only `<a>` link anchors and `<input>` fields are selected, whereas `qutebrowser` gives me the impression of selecting more stuff.

These differences may seem minute, but they are important for having the `lynx` experience.
I like the idea of having a mode, although a lot of `lynx`-lovers (including myself) would crave to have ***an option to switch this mode `on` by default.**

In conclusion, it would not be a bad idea to reopen this request. Just to keep track of it.

**Comment by amacfie** [comment]
Just enter insert mode and use tab, shift-tab, and enter?

**Comment by stroobandt** [comment]
Sorry, I cannot test since I am no longer using the browser…

**Comment by The-Compiler** [comment]
Yeah, I think using `<Tab>` is pretty close actually - also see #1305.

**Comment by amacfie** [comment]
See also https://github.com/qutebrowser/qutebrowser/issues/5033
